ambry-network/src/test/java/com.github.ambry.network/SSLSelectorTest.java

```diff
@@ -203,6 +202,36 @@ public void testEmptyRequest()
     assertEquals("", blockingRequest(connectionId, ""));
   }
 
+  @Test
+  public void testSSLConnect()
+      throws IOException {
+    String connectionId =
+        selector.connect(new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE, PortType.SSL);
+    Assert.assertFalse("Channel should not be ready by now (until handshake completes)",
+        selector.isChannelReady(connectionId));
+    while (!selector.connected().contains(connectionId)) {
+      selector.poll(10000L);
+    }
+    Assert.assertTrue("Channel should have been ready by now ", selector.isChannelReady(connectionId));
+  }
+
+  @Test
+  public void testCloseDuringSslHandshake()
+      throws IOException {
+    String connectionId =
+        selector.connect(new InetSocketAddress("localhost", server.port), BUFFER_SIZE, BUFFER_SIZE, PortType.SSL);
+    Assert.assertFalse("Channel should not be ready by now (until handshake completes)",
+        selector.isChannelReady(connectionId));
+    selector.poll(10000L);
+    Assert.assertFalse("Channel shouldn't have been added to connect list w/o handshake",
+        selector.connected().contains(connectionId));
+    selector.close(connectionId);
+    Assert.assertFalse("Channel should not have been added to connected list ",
```

Comment:
I meant about the `assert` that `connected()` would contain the `connectionId` after the `close()`. Things do not asynchronously get populated to the `connected()` list. The `selector` has to be polled for it to put connections established into the `connected()` list. So the second assert in the following is pointless and misleading and is what you are doing above:

```java
assertFalse(selector.connected().contains(connId));
selector.close();
assertFalse(selector.connected().contains(connId));
```